src/test/java/hudson/security/LDAPEmbeddedFIPSTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2017 CloudBees,Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package hudson.security;
+
+import hudson.util.FormValidation;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import jenkins.model.Jenkins;
+import hudson.tasks.MailAddressResolver;
+import hudson.util.Secret;
+
+import java.security.cert.X509Certificate;
+
+import jenkins.model.IdStrategy;
+import jenkins.security.plugins.ldap.*;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.rules.ExpectedException;
+import org.jvnet.hudson.test.FlagRule;
+import org.springframework.security.ldap.userdetails.LdapUserDetails;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+
+import javax.net.ssl.*;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+
+@LDAPTestConfiguration(ldapsProtocol = true)
+public class LDAPEmbeddedFIPSTest {
+    public LDAPRule ads = new LDAPRule();
```

Comment:
I doubt that you need an actual ldap server running in the background for the tests you are going to do, it will only take up unnecessary compute.

Reply:
I'm just trying to login using the secure ldap url .